### DISCUSSION TITLE
Add pagination and filtering to Queue Manager

### DISCUSSION THREAD

**Discussion by soulseekah** [craftcms/cms#14755] [Ideas] [answered]
We have a lot of delayed jobs in the list, so the queue manager is quite useless.

<img width="805" alt="image" src="https://github.com/craftcms/cms/assets/685880/5099fc1c-2024-4875-a5cd-ed5d3e7de364">

The screen fetched only the latest 50 added jobs. This is not enough to find failed jobs in the list and check their errors, rerun some.

We have a workaround using the following:

```php
Event::on(
    BaseController::class,
    BaseController::EVENT_BEFORE_ACTION,
    static function (ActionEvent $event) {
        /**
         * Output more jobs in the queue manager admin screen
         * by overriding the limit POST/GET parameter in the Request.
         */
        if ('queue/get-job-info' === $event->action->uniqueId) {
            $limit = 5000; // The limit of jobs on the queue manget screen.
            $params = Craft::$app->request->getBodyParams();
            $params['limit'] = $limit;
            Craft::$app->request->setBodyParams($params);

            $params = Craft::$app->request->getQueryParams();
            $params['limit'] = $limit;
            Craft::$app->request->setQueryParams($params);
        }
    }
);
```

This outputs 5000 jobs on the screen which we can then search through for failed jobs in the browser. This is a hack.

It would be nice to have pagination and filtering on the Queue Manager screen instead.

**Comment by brandonkelly** [comment, marked as answer]
Adding an `offset` argument to `QueueInterface::getJobInfo()` (which would be required for pagination) would be a breaking change, so not something we could consider until Craft 6.

However I’ve just fixed the built-in queue so it lists delayed jobs after other non-delayed jobs, which should help you here. (19df31b53790ce55ae62850789fad7860f39122d)

**Reply by brandonkelly** [reply]
Craft 5.0.2 is out with that fix. (It’s fixed for the next 4.x release as well, whenever that comes out…)

**Reply by soulseekah** [reply]
Noice! Thanks!

**Reply by brandonkelly** [reply]
4.8.8 is also out now with that fix.

**Comment by bymayo** [comment]
Just to add to this, i remember there was a Craft CMS 2 (?!) plugin that handled the queue before it was part of Craft itself. 

That plugin had an option to filter down the tasks by type.

Similar to this issue we have 700 failed tasks that have built up overtime, but im weary of deleting them all (Mostly Blitz cache tasks) incase theres more important tasks mixed in e.g. Sending email notifications etc. 

Normally I do this via the database, but it would be easier to filter tasks down in the CP, with the ability to delete/release groups of tasks.

Pagination or infinite scroll would be great also.
